### PR TITLE
feat: Replace raw! and styled! with span! macro

### PR DESCRIPTION
**Pull request by kdheepak**

### CODE REVIEW

File: src/line.rs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+///
+/// ```rust
+/// # use ratatui::prelude::*;
+/// use ratatui_macros::line;
+///
+/// let line = line!["hello", "world"];
+/// let line = line!["hello".red(), "world".red().bold()];
+/// ```
+///
+/// * Create a [`Line`] from a given [`Span`] and size:
```
Comment (joshka) [inline src/line.rs:17]: comment from other PR
```suggestion
/// * Create a [`Line`] that repeats a [`Span`] a given amount of times:
```

File: src/span.rs [hunk 2]
```diff
@@ -1,74 +1,55 @@
-/// A macro for creating a raw (unstyled) [`Span`] using formatting syntax.
+/// A macro for creating a [`Span`] using formatting syntax.
 ///
-/// `raw!` is similar to the [`format!`] macro, but it returns a [`Span`] instead of a `String`.
+/// `span!` is similar to the [`format!`] macro, but it returns a [`Span`] instead of a `String`.
+///
+/// The first argument is a [`Style`] or any type that is convertible to [`Style`] (e.g. [`Color`]).
```
Comment (joshka) [inline src/span.rs:5]: "If there is a semicolon, then ... otherwise, the Span will be created as a raw span (with style set to Style::default()).

**Review by joshka — APPROVED** [review]

File: src/span.rs (hunk 2 above)
Comment (kdheepak) [reply src/span.rs]: Resolved in https://github.com/ratatui-org/ratatui-macros/pull/30/commits/f6cea575bcac9f289fb3d1c6a44e057cbeea795e

**Comment by kdheepak** [comment]
This was technically a breaking change. 

Should I rewrite history or add a new empty commit with that links to this commit indicating that this was a breaking change?

**Comment by kdheepak** [comment]
I haven't been diligent with the commit history here, so it's not particularly useful.

**Pull request merged**